custom-steps/expat/post-build.ps1

```diff
@@ -1,13 +0,0 @@
-param (
-    [Parameter(Mandatory=$true)][string]$BuildArtifactsPath
-)
-
-Import-Module "$PSScriptRoot/../../ps-modules/Build" -DisableNameChecking -Force
-
-if ((Get-IsOnMacOS)) {
-    Remove-DylibSymlinks -BuildArtifactsPath $BuildArtifactsPath
-}
-elseif((Get-IsOnWindowsOS)) {
-    Update-VersionInfoForDlls -buildArtifactsPath $buildArtifactsPath -versionInfoJsonPath "$PSScriptRoot/version-info.json"
-}
```

Comment:
It looks like they fixed the version number issue, but the final dll is still missing metadata for:
- File Description
- Product Name
- Copyright

![image](https://github.com/TechSmith/ThirdParty-Packages-vcpkg/assets/62569405/d4732729-1739-4b9a-a32e-9ed32ba0c844)

Reply:
I think it is useful for security scans to have at least those File description and Product name fields?

One option that I think would work (would have to test it) is to just remove the fileVersion and productVersion elements from the version-info.json and just keep the rest in there.  I think it's smart enough to only update the metadata for the elements that are in that json file.

Reply:
I personally don't want to add extra step. For example,  later when expat fill those metadatas,  our custom step will overrite them.  But if you have strong opinion,  I can try what you suggested,  but I still want to remove those hard coded 2.5.0 in File description and Product name fields,  because Dev won't remember to update this file everytime updating Expat.

Reply:
with the [tmp commit](https://github.com/TechSmith/ThirdParty-Packages-vcpkg/pull/22/commits/ad3a57b3931d30b109dea953e11cba9397c71963),  I verified I can get:

![image](https://github.com/TechSmith/ThirdParty-Packages-vcpkg/assets/5375159/20820dbb-89b8-4059-b6f5-7c071bf35217)

Still, I kind of want to get rid of this step.  please let me know

Reply:
> RE: Updating non-version metadata independently (https://github.com/TechSmith/ThirdParty-Packages-vcpkg/pull/22/commits/ad3a57b3931d30b109dea953e11cba9397c71963)

I think it might be the right decision to do something like you have in that commit, for now.

This is kind of a pattern that we are accepting will have to be corrected with a manual step for some of the other packages we build with this (4 others): https://github.com/search?q=repo%3ATechSmith%2FThirdParty-Packages-vcpkg%20version-info.json&type=code.

> I want to get rid of this step
I definitely do, too.  I think there are two possible reasons we might need to keep this metadata, though:
1. To support security-scanning tools
2. To comply with open source licenses

## 1. To support security-scanning tools
I'm pretty sure FileVersion is used by security scanning tools, but I'm actually less certain about these other metadata fields, so I asked in #dependency-security:  https://techsmith.slack.com/archives/C01TV8BD1GA/p1709579610207789

## 2. To comply with open source licenses
I'm actually not sure that we technically need this copyright info in the dll metadata on Windows or not.  I think we are probably covered by the fact that we [distribute copyright & license info with our apps](https://techsmith.slack.com/archives/C01TV8BD1GA/p1691601655012979?thread_ts=1691595673.860119&cid=C01TV8BD1GA):
- CamtasiaWin: `Doc/3rdPartyLicensing.txt`
- CamtasiaMac: `CamtasiaGo/Resources/Credits.rtf`
- SnagitWin: `Docs/ENU/3rdParty_Licenses.txt`
- SnagitMac: `SmacitEditor/en.lproj/Acknowledgments.rtf`

I've added them with rcedit like this just to be safe to the other dlls we build through ThirdParty-Packages-vcpkg, however, when they are missing.

Reply:
Anyway, all that being said, it's possible it's not technically needed (beyond FileVersion), but I'd personally feel more comfortable if we just kept all that metadata in there, even if it does mean that we have a manual step like this to do anytime we update the libs that have a `version-info.json` in the custom-steps directory for it.

Reply:
Hmm,  OK,  I made the commit official.

Reply:
Yep, it's a manual step we'll have to do for now until they get that sorted out.

This might be a good opportunity for some TechSmith devs to even contribute upstream patches to fix these vcpkg ports that don't pull in all the metadata appropriately.  I intend to take a look at this in a future IP sprint.  It might be pretty straightforward to fix.